editor/components/inserter/menu.js

```diff
@@ -12,10 +12,161 @@ import Dashicon from 'components/dashicon';
+				focusables.push( refName );
+			}
+		}
+
+		// Add search regardless at the end.
+		focusables.push( 'search' );
+
+		return focusables;
+	}
+
+	nextFocusableRef( component ) {
+		const focusables = this.visibleFocusables( component.nodes, component.state.filterValue );
+
+		// Initiate the menu with the first block.
+		if ( null === component.state.focusedElementRef ) {
+			return focusables[ 0 ];
+		}
+
+		const currentIndex = focusables.findIndex( ( elementRef ) => elementRef === component.state.focusedElementRef );
+		return this.nextFocusInList( currentIndex, focusables );
+	}
+
+	previousFocusableRef( component ) {
+		const focusables = this.visibleFocusables( component.nodes, component.state.filterValue );
+
+		// Initiate the menu with the first block.
+		if ( null === component.state.focusedElementRef ) {
+			return focusables[ 0 ];
+		}
+
+		const currentIndex = focusables.findIndex( ( elementRef ) => elementRef === component.state.focusedElementRef );
+		return this.previousFocusInList( currentIndex, focusables );
+	}
+
+	/**
+	 * Get the next focusable element in list.
+	 *
+	 * @param {int} currentIndex Current index in the list.
+	 * @param {array} focusables List of focusable elements attached to node references.
+	 * @return {string} Name of the next focus.
+	 */
+	nextFocusInList( currentIndex, focusables ) {
+		const nextIndex = currentIndex + 1;
+		const highestIndex = focusables.length - 1;
+
+		// Check boundary so that the index is does not exceed the length.
+		if ( nextIndex > highestIndex ) {
+			// Cycle back to other end.
+			return focusables[ 0 ];
+		}
+
+		return focusables[ nextIndex ];
+	}
+
+	/**
+	 * Get the previousFocusInList.
+	 *
+	 * @param {int} currentIndex Current index in the list.
+	 * @param {array} focusables List of focusable elements attached to node references.
+	 * @return {string} Name of the previous focus.
+	 */
+	previousFocusInList( currentIndex, focusables ) {
+		const nextIndex = currentIndex - 1;
+		const lowestIndex = 0;
+
+		// Check boundary so that the index is does not exceed the length.
+		if ( nextIndex < lowestIndex ) {
+			// Cycle back to other end.
+			return focusables[ focusables.length - 1 ];
+		}
+
+		return focusables[ nextIndex ];
+	}
+
+	focusNextFocusableRef( component ) {
+		const next = component.nextFocusableRef( component );
+		component.changeMenuSelection( next );
+	}
+
+	focusPreviousFocusableRef( component ) {
+		const previous = component.previousFocusableRef( component );
+		component.changeMenuSelection( previous );
+	}
+
+	listVisibleBlocksByCategory( blocks ) {
+		return blocks.reduce( ( groups, block ) => {
+			if ( ! this.isShownBlock( this.state.filterValue, block ) ) {
+				return groups;
+			}
+			if ( ! groups[ block.category ] ) {
+				groups[ block.category ] = [];
+			}
+			groups[ block.category ].push( block );
+			return groups;
+		}, {} );
+	}
+
+	bindReferenceNode( nodeName ) {
```

Comment:
This is used to also bind the search field.  The search field of the inserter is not a blockType so I used nodeName instead is blockType preferred?